### PR TITLE
Close streams which are created in PrintCSV

### PR DESCRIPTION
**Pull request by ChrisJefferson**
Code which calls PrintCSV many times with a filename will eventually start failing, as PrintCSV did not close files after it was finished with them.

### CODE REVIEW

**Comment by fingolfin** [comment]
It's really a shame that we don't have finalizers to take care of this!

File: tst/testinstall/stringobj.tst [hunk 1]
```diff
@@ -119,10 +119,11 @@ gap> s := ReadCSV( filename{[1..Length(filename)-4]}, ";" );
...
   rec( f := 11, f_2 := 12 ), rec( f := "yy", f1 := "x", f_2 := "zzz" ) ]
 gap> tmpdir := DirectoryTemporary();;
-gap> fname := Filename(tmpdir, "output.csv");;
-gap> PrintCSV( fname, s );
-gap> s = ReadCSV( fname );
-true
+gap> for i in [1..1000] do
+>   fname := Filename(tmpdir, "output.csv");;
+>   PrintCSV( fname, s );
+>   if s <> ReadCSV( fname ) then Print("failed loop", i,"\n"); fi;
```
Comment (fingolfin) [inline tst/testinstall/stringobj.tst]: If I look at this test in a year, I won't have a clue what it is about. I'd have to dig through the commit history. Perhaps a brief comment to explain that could be added?

Comment (ChrisJefferson) [reply tst/testinstall/stringobj.tst]: Good idea, now fixed

**Comment by ChrisJefferson** [comment]
I agree the lack of finalisers is annoying, in particular this is still "bugged" if we exit due to an error early, but in current GAP that's basically unfixable.

**Pull request merged**